eodatasets3/validate.py

```diff
@@ -615,7 +615,10 @@ def numpy_value_fits_dtype(value, dtype):
     if _is_nan(value):
         return np.issubdtype(dtype, np.floating)
     else:
-        return np.all(np.array([value], dtype=dtype) == [value])
+        try:
+            return np.all(np.array([value], dtype=dtype) == [value])
+        except Exception:  # pylint: disable=broad-except
```

Comment:
I guess this will work, but it seems a bit... heavy handed.

Reply:
I was getting an `OverflowError` in a test, but wasn't sure if that was the only kind of error that might be produced here.